docs/intro/contributing.txt

```diff
@@ -103,26 +116,101 @@ Download the Django source code repository using the following command:
 
     $ git clone https://github.com/django/django.git
 
+Now that you have a local copy of Django, you can install it just like you would
+install any package using ``pip``. The most convenient way to do so is by using
+a *virtual environment* (or virtualenv) which is a feature built into Python
+that allows you to keep a separate directory of installed packages for each of
+your projects so that they don't interfere with each other.
+
+It's a good idea to keep all your virtualenvs in one place, for example in
+``.virtualenvs/`` in your home directory. Create it if it doesn't exist yet:
+
+.. code-block:: console
+
+    $ mkdir ~/.virtualenvs
+
+Now create a new virtualenv by running:
+
+.. code-block:: console
+
+    $ python3 -m venv ~/.virtualenvs/djangodev
+
+The path is where the new environment will be saved on your computer.
+
+If you are on Windows, run these commands in the Git Bash shell:
+
+.. code-block:: none
+
+    > mkdir C:Users/name/virtualenvs
```

Comment:
Yes it is a bit odd. I will set everything up again from scratch on a new machine and type every single line verbatim (without e.g. tab completion) to make sure these instructions work 100%.